apps/server/src/shared/testing/factory/domainobject/tool/external-tool-datasheet-template-data.factory.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	CustomParameterLocation,
+	LtiMessageType,
+	LtiPrivacyPermission,
+	ToolConfigType,
+} from '@modules/tool/common/enum';
+import { DeepPartial, Factory } from 'fishery';
+import {
+	ExternalToolDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateData,
+} from '@modules/tool/external-tool/domain';
+
+export const externalToolParameterDatasheetTemplateDataFactory = Factory.define<
+	ExternalToolParameterDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateData
+>(({ sequence }) => {
+	return {
+		name: `custom-parameter-${sequence}`,
+		properties: '',
+		type: 'Zeichenkette',
+		scope: 'Schule',
+		location: CustomParameterLocation.BODY,
+	};
+});
+
+export class ExternalToolDatasheetTemplateDataFactory extends Factory<ExternalToolDatasheetTemplateData> {
+	asOauth2Tool(): this {
+		const params: DeepPartial<ExternalToolDatasheetTemplateData> = {
+			toolType: 'OAuth 2.0',
+			skipConsent: 'ja',
+			toolUrl: 'https://www.oauth2-baseUrl.com/',
+		};
+		return this.params(params);
+	}
+
+	asLti11Tool(): this {
+		const params: DeepPartial<ExternalToolDatasheetTemplateData> = {
+			toolType: 'LTI 1.1',
+			messageType: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+			privacy: LtiPrivacyPermission.PSEUDONYMOUS,
+			toolUrl: 'https://www.lti11-baseUrl.com/',
+		};
+		return this.params(params);
+	}
+
+	withParameters(number: number, customParam?: DeepPartial<ExternalToolParameterDatasheetTemplateData>): this {
+		const params: DeepPartial<ExternalToolDatasheetTemplateData> = {
+			parameters: externalToolParameterDatasheetTemplateDataFactory.buildList(number, customParam),
+		};
+		return this.params(params);
+	}
+
+	withOptionalParameters(): this {
```

Comment:
optional parameters are something different in the context of external tools. please choose different name.